pkg/chains/formats/slsa/v2alpha2/internal/pipelinerun/pipelinerun.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2023 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pipelinerun
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	intoto "github.com/in-toto/in-toto-golang/in_toto"
+	slsa "github.com/in-toto/in-toto-golang/in_toto/slsa_provenance/v1"
+	"github.com/tektoncd/chains/pkg/chains/formats/slsa/extract"
+	resolveddependencies "github.com/tektoncd/chains/pkg/chains/formats/slsa/v2alpha2/internal/resolved_dependencies"
+	"github.com/tektoncd/chains/pkg/chains/objects"
+)
+
+const pipelineRunResults = "pipelineRunResults/%s"
+
+// GenerateAttestation generates a provenance statement with SLSA v1.0 predicate for a pipeline run.
+func GenerateAttestation(ctx context.Context, builderID string, pro *objects.PipelineRunObject) (interface{}, error) {
+	rd, err := resolveddependencies.PipelineRun(ctx, pro)
+	if err != nil {
+		return nil, err
+	}
+	bp, err := byProducts(pro)
```

Comment:
NIT: `byproducts` is a single word. No need to capitalize `P` here.

Reply:
Ah yes. Will fix.

Reply:
Ah yes.